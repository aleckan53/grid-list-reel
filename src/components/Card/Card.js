import React from 'react';
import styles from './Card.module.css';

export default function Card() {

  const bgImage = {
    backgroundImage: `url(${data.img})`,
  };

  return (
    <li className={styles.card}>

      <div className={ styles.img } style={ bgImage }/>
      <div className={ styles.info }>
        <h3>{ data.name }</h3>
        <p>{ data.saved }</p>
        <p>{ data.status }</p>
      </div>
      <div className={ styles.progress }>
        { data.percentSaved }
      </div>
      <div className={ styles.details }></div>
      <button className={ styles.toggle }>View details</button>
    </li>
  )
}


const data = {
  img: 'http://www.pngmart.com/files/3/BMW-M3-Transparent-Background.png',
  name: 'BMW 328ci',
  status: 'active',
  total: 35000,
  started: '2017-5-1',
  perDay: 15,
  saved: 12000,
  percentSaved: 34,
}