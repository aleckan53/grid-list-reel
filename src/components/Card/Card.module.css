.card {
  list-style: none;
  background: white;
  border: var(--border) var(--black30);
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: 25% auto 25%;
      grid-template-columns: 25% auto 25%;
  -ms-grid-rows: auto 0% 20%;
      grid-template-rows: auto 0% 20%;
  grid-column-gap: 1rem;
  grid-row-gap: .5rem;
  padding: 1rem;
      grid-template-areas: "img info progress" "details details details" ". expand .";
}

.card .img {
  -ms-grid-row: 1;
  -ms-grid-column: 1;
  grid-area: img;
  border: var(--border) var(--black30);
}

.card .info {
  -ms-grid-row: 1;
  -ms-grid-column: 2;
  grid-area: info;
}

.card .progress {
  -ms-grid-row: 1;
  -ms-grid-column: 3;
  grid-area: progress;
}

.card .details {
  -ms-grid-row: 2;
  -ms-grid-column: 1;
  -ms-grid-column-span: 3;
  grid-area: details;
  height: 500px;
  width: 100%;
  display: none;
}

.card .toggle {
  -ms-grid-row: 3;
  -ms-grid-column: 2;
  grid-area: expand;
  text-align: center;
}
/*# sourceMappingURL=Card.module.css.map */