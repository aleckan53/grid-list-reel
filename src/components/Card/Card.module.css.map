{
    "version": 3,
    "mappings": "AAGA,AAAA,KAAK,CAAC;EACJ,UAAU,EAAE,IAAI;EAChB,UAAU,EAAE,KAAK;EACjB,MAAM,EALC,aAAa,CADZ,cAAc;EAQtB,OAAO,EAAE,IAAI;EACb,qBAAqB,EAAE,YAAY;EACnC,kBAAkB,EAAE,WAAW;EAC/B,eAAe,EAAE,IAAI;EACrB,YAAY,EAAE,KAAK;EACnB,OAAO,EAAE,IAAI;EACb,mBAAmB,EAAE,0DAEY;CA0BlC;;AAvCD,AAeE,KAfG,CAeH,IAAI,CAAC;EACH,SAAS,EAAE,GAAG;EACd,MAAM,EAnBD,aAAa,CADZ,cAAc;CAqBrB;;AAlBH,AAoBE,KApBG,CAoBH,KAAK,CAAC;EACJ,SAAS,EAAE,IAAI;CAChB;;AAtBH,AAwBE,KAxBG,CAwBH,SAAS,CAAC;EACR,SAAS,EAAE,QAAQ;CACpB;;AA1BH,AA4BE,KA5BG,CA4BH,QAAQ,CAAC;EACP,SAAS,EAAE,OAAO;EAClB,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,IAAI;CACd;;AAjCH,AAmCE,KAnCG,CAmCH,OAAO,CAAC;EACN,SAAS,EAAE,MAAM;EACjB,UAAU,EAAE,MAAM;CACnB",
    "sources": [
        "Card.module.scss"
    ],
    "names": [],
    "file": "Card.module.css"
}