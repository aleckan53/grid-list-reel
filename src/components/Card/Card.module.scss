$black30: var(--black30);
$border: var(--border);

.card {
  list-style: none;
  background: white;
  border: $border $black30;

  display: grid;
  grid-template-columns: 25% auto 25%;
  grid-template-rows: auto 0% 20%;
  grid-column-gap: 1rem;
  grid-row-gap: .5rem;
  padding: 1rem;
  grid-template-areas: "img info progress"
                       "details details details"
                       ". expand .";

  .img {
    grid-area: img;
    border: $border $black30;
  }

  .info {
    grid-area: info;
  }

  .progress {
    grid-area: progress;
  }

  .details {
    grid-area: details;
    height: 500px;
    width: 100%;
    display: none;
  }

  .toggle {
    grid-area: expand;
    text-align: center;
  }
}