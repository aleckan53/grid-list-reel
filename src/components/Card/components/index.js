import Image from './Image';
import Info from './Info';
import Progress from './Progress';
import Details from './Details';

export {
  Image,
  Info,
  Progress,
  Details
};