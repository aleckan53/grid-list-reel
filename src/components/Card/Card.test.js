import React from 'react';
import ReactDOM from 'react-dom';

import Card from './Card';

describe('Card component', () => {
  let root = document.createElement('div');

  beforeEach(() => {
    ReactDOM.render(<Card />, root);
  })

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(root);
  })

  it('matches the snapshot', () => {
    expect(root).toMatchSnapshot();
  })
    
});