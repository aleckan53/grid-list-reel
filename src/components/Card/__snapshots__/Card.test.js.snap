// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches the snapshot 1`] = `
<div>
  <li
    class="card"
  >
    <div
      class="img"
      style="background-image: url(http://www.pngmart.com/files/3/BMW-M3-Transparent-Background.png);"
    />
    <div
      class="info"
    >
      <h3>
        BMW 328ci
      </h3>
      <p>
        12000
      </p>
      <p>
        active
      </p>
    </div>
    <div
      class="progress"
    >
      34
    </div>
    <div
      class="details"
    />
    <button
      class="toggle"
    >
      View details
    </button>
  </li>
</div>
`;
